kpatch-build/create-klp-module.c

```diff
@@ -292,6 +293,21 @@ static void create_klp_arch_sections(struct kpatch_elf *kelf, char *strings)
 		 * single .klp.arch.vmlinux..parainstructions section
 		 */
 		old_size = sec->data->d_size;
+
+		/*
+		 * Due to a quirk in how .parainstructions gets linked, the
+		 * section size doesn't encompass the last 4 bytes of the last
+		 * entry.  Align the old size properly before merging.
+		 */
+		if (!strcmp(base->name, ".parainstructions")) {
+			str = getenv("PARA_STRUCT_SIZE");
+			if (!str)
+				ERROR("PARA_STRUCT_SIZE not set");
+			align_mask = atoi(str) - 1;
+
```

Comment:
Up to you, but you could theoretically cache the align_mask value once you've calculated it to save extra getenv() + atoi() lookups.  The surrounding code and the math below looks good to me.